mm2src/mm2_main/tests/docker_tests/swap_watcher_tests.rs

```diff
@@ -0,0 +1,965 @@
+}
+
+fn get_balance_f64(mm_node: &MarketMakerIt, ticker: &str) -> f64 {
+    get_balance(mm_node, ticker).parse::<f64>().unwrap()
+}
+
+#[test]
+#[ignore]
+fn test_watcher_spends_maker_payment_spend_eth_erc20() {
+    let coins = json!([eth_testnet_conf(), eth_jst_conf(ETH_SEPOLIA_TOKEN_CONTRACT)]);
+
+    let alice_passphrase =
+        String::from("spice describe gravity federal blast come thank unfair canal monkey style afraid");
+    let alice_conf = Mm2TestConf::seednode_using_watchers(&alice_passphrase, &coins);
+    let mut mm_alice = MarketMakerIt::start(alice_conf.conf.clone(), alice_conf.rpc_password.clone(), None).unwrap();
+    let (_alice_dump_log, _alice_dump_dashboard) = mm_alice.mm_dump();
+    log!("Alice log path: {}", mm_alice.log_path.display());
+
+    let bob_passphrase = String::from("also shoot benefit prefer juice shell elder veteran woman mimic image kidney");
+    let bob_conf = Mm2TestConf::light_node(&bob_passphrase, &coins, &[&mm_alice.ip.to_string()]);
+    let mut mm_bob = MarketMakerIt::start(bob_conf.conf, bob_conf.rpc_password, None).unwrap();
+    let (_bob_dump_log, _bob_dump_dashboard) = mm_bob.mm_dump();
+    log!("Bob log path: {}", mm_bob.log_path.display());
+
+    let watcher_passphrase =
+        String::from("also shoot benefit prefer juice shell thank unfair canal monkey style afraid");
+    let watcher_conf = Mm2TestConf::watcher_light_node(
+        &watcher_passphrase,
+        &coins,
+        &[&mm_alice.ip.to_string()],
+        0.,
+        1.5,
+        1.,
+        0.,
+    )
+    .conf;
+    let mut mm_watcher = MarketMakerIt::start(watcher_conf, "pass".to_string(), None).unwrap();
+    let (_watcher_dump_log, _watcher_dump_dashboard) = mm_dump(&mm_watcher.log_path);
+
+    enable_eth_and_jst(&mm_alice);
+    enable_eth_and_jst(&mm_bob);
+    enable_eth_and_jst(&mm_watcher);
+
+    let alice_eth_balance_before = get_balance_f64(&mm_alice, "ETH");
+    let alice_jst_balance_before = get_balance_f64(&mm_alice, "JST");
+    let bob_eth_balance_before = get_balance_f64(&mm_bob, "ETH");
+    let bob_jst_balance_before = get_balance_f64(&mm_bob, "JST");
+
+    block_on(start_swaps(&mut mm_bob, &mut mm_alice, &[("ETH", "JST")], 1., 1., 0.01));
+
+    block_on(mm_alice.wait_for_log(180., |log| log.contains(WATCHER_MESSAGE_SENT_LOG))).unwrap();
+    block_on(mm_alice.stop()).unwrap();
+    block_on(mm_watcher.wait_for_log(180., |log| log.contains(MAKER_PAYMENT_SPEND_SENT_LOG))).unwrap();
+    thread::sleep(Duration::from_secs(25));
+
+    let mm_alice = MarketMakerIt::start(alice_conf.conf.clone(), alice_conf.rpc_password.clone(), None).unwrap();
+    enable_eth_and_jst(&mm_alice);
+
+    let alice_eth_balance_after = get_balance_f64(&mm_alice, "ETH");
+    let alice_jst_balance_after = get_balance_f64(&mm_alice, "JST");
+    let bob_eth_balance_after = get_balance_f64(&mm_bob, "ETH");
+    let bob_jst_balance_after = get_balance_f64(&mm_bob, "JST");
```

Comment:
`BigDecimal` is preferred over `f64` in almost all cases due to floating-point errors.